### PR TITLE
fix: remove local copy of beam I/O now that it is merged to upstream

### PR DESCRIPTION
**Pull request by dpcollins-google**

### CODE REVIEW

**Review by anguillanneuf — CHANGES_REQUESTED** [review]
- Please also remove `pubsublite-beam-io` from `versions.txt`.
- Is the PubsubLiteIO in Beam version `2.32.0` equivalent to version `0.19.5` here?  If not, we may consider removing the files here after the next Beam release.
- Please also resolve the merge conflicts. Thank you!
- Please also update this JIRA ticket: https://issues.apache.org/jira/browse/BEAM-10114
- (FYI) sample has been updated: https://github.com/GoogleCloudPlatform/java-docs-samples/pull/5312. Docs too. (cl/396653855)

File: samples/snapshot/pom.xml [hunk 1]
```diff
@@ -49,23 +48,6 @@
...
-      <groupId>com.google.cloud</groupId>
-      <artifactId>pubsublite-beam-io</artifactId>
-      <version>0.19.6-SNAPSHOT</version>
-    </dependency>
-    <!-- {x-version-update-end} -->
-    <dependency>
-      <groupId>com.google.cloud</groupId>
-      <artifactId>google-cloud-pubsub</artifactId>
-      <version>1.114.3</version>
-    </dependency>
```
[the remark this answers is not in the document]
Comment (dpcollins-google) [reply samples/snapshot/pom.xml]: Done.

File: pubsublite-beam-io/README.md [hunk 2]
```diff
@@ -1,87 +0,0 @@
...
-
-    >`java.lang.IllegalAccessError: failed to access class com.google.cloud.
-    pubsublite.internal.wire.RoutingMetadata from class com.google.cloud.
-    pubsublite.beam.SubscriberOptions (com.google.cloud.pubsublite.internal.
-    wire.RoutingMetadata and com.google.cloud.pubsublite.beam.SubscriberOptions
-    are in unnamed module of loader java.net.URLClassLoader @ff2266c)`
-
-1. Aggregate transforms after fixed windowing will not emit elements in the
-DirectRunner for Pub/Sub Lite I/O unless you explicitly fire after watermark
-triggers, or use processing-time triggers instead.
```
[the remark this answers is not in the document]
Comment (dpcollins-google) [reply pubsublite-beam-io/README.md]: I've left the file with its contents but updated them to be correct for the version that will be released in 2.33.0. Do you want me to leave the text or replace it with a reference as described above?

**Comment by dpcollins-google** [comment]
> * Please also remove `pubsublite-beam-io` from `versions.txt`.

Done

> * Is the PubsubLiteIO in Beam version `2.32.0` equivalent to version `0.19.5` here?  If not, we may consider removing the files here after the next Beam release.

Version 2.33.0 will be equivalent, I'll add a DO_NOT_MERGE label for now.

> * Please also resolve the merge conflicts. Thank you!

Done

> * Please also update this JIRA ticket: https://issues.apache.org/jira/browse/BEAM-10114

Done

> * (FYI) sample has been updated: [GoogleCloudPlatform/java-docs-samples#5312](https://github.com/GoogleCloudPlatform/java-docs-samples/pull/5312). Docs too. (cl/396653855)

File: pubsublite-beam-io/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (anguillanneuf) [reply pubsublite-beam-io/README.md]: We should add the above reference in the top. Keeping your changes means we need to keep it updated. We can do that for the next release and then remove.

[the remark this answers is not in the document]
Comment (dpcollins-google) [reply pubsublite-beam-io/README.md]: Done.

**Review by manuelmenzella-google — APPROVED** [review]

**Pull request merged**